pyomo/repn/plugins/gams_writer.py

```diff
@@ -327,29 +448,20 @@ def _write_model(self,
         linear = True
         linear_degree = set([0,1])
 
+        # Make sure there are no strange ActiveComponents. The expression
+        # walker will handle strange things in constraints later.
         model_ctypes = model.collect_ctypes(active=True)
-        if False:
-            #
-            # WEH - Disabling this check.  For now, we're allowing
-            # variables defined on non-block objects.
-            #
-            # Sanity check: all active components better be things we know
-            # how to deal with, plus Suffix if solving
-            valid_ctypes = set([
-                Block, Constraint, Expression, Objective, Param,
-                Set, RangeSet, Var, Suffix, Connector ])
-            if not model_ctypes.issubset(valid_ctypes):
-                invalids = [t.__name__ for t in (model_ctypes - valid_ctypes)]
-                raise RuntimeError(
-                    "Unallowable component(s) %s.\nThe GAMS writer cannot "
-                    "export models with this component type" %
-                    ", ".join(invalids))
-
-        # HACK: Temporary check for Connectors in active constriants.
-        # This should be removed after the writer is moved to an
-        # explicit GAMS-specific expression walker for generating the
-        # constraint strings.
-        has_Connectors = Connector in model_ctypes
+        valid_ctypes = set([Block, Constraint, Objective, Suffix])
+        invalids = set()
+        for t in (model_ctypes - valid_ctypes):
+            if issubclass(t, ActiveComponent):
+                invalids.add(t)
```

Comment:
It is not invalid to have a Component derived from an ActiveComponent on the model, as long as it is deactivated.  I think you need to go through your list of ctypes here and collect the list of active components.  If that list is empty, all is good -- otherwise throw the error.

Reply:
I think this is already doing what you want. I use `model_ctypes = model.collect_ctypes(active=True)` to collect all the active ctypes on the model, and I subtract `valid_ctypes` which is the set of active ctypes I am ok with. Then I go through what remains and if anything in there is an `ActiveComponent`, that means there is an active component on the model that is not deactivated and should be, so I throw an error.